RecoTracker/TkHitPairs/src/RecHitsSortedInPhi.cc

```diff
@@ -21,8 +22,15 @@ RecHitsSortedInPhi::RecHitsSortedInPhi(const std::vector<Hit>& hits, GlobalPoint
   // assert(origin.x()==0 && origin.y()==0);
 
   theHits.reserve(hits.size());
-  for (auto const& hp : hits)
+  for (auto const& hp : hits) {
+    auto gp = hp->globalPosition();
```

Comment:
this is quite expensive and unnecessary as NaN is impossible to happen.
If one protect here should protect in some thousand other places

I suggest to leave it as it was.